mariamihaelafainarea/assignment3/Assignment3Test

```diff
@@ -0,0 +1,226 @@
+        Iterator<String> it = s.iterator();
+        StringBuilder bigString = new StringBuilder("");
+        while (it.hasNext()) {            
+            String currentString = it.next();
+            System.out.println(currentString);
+            bigString.append(currentString);
+            bigString.append(" ");
+        }
+        bigString.deleteCharAt(bigString.length()-1);
+        System.out.println(bigString);
+        assertEquals(bigString.toString(),"ader aer amar apa apar");
+        
+        
+    } 
+     @Test
+    public void EmptyDictionaryTest(){
+        // we have nothing in the dictionary
+        //so we cannot form anything
+        Assignment3 b = new Assignment3();
+        Assignment3.MyDictionary dictionary =  b.new MyDictionary();
+        char[][] grid = new char[3][3];
+        grid[0][0] = 'd';
+        grid[0][1] = 'c';
+        grid[0][2] = 'b';
+        grid[1][0] = 'e';
+        grid[1][1] = 'a';
+        grid[1][2] = 'p';
+        grid[2][0] = 'r';
+        grid[2][1] = 'a';
+        grid[2][2] = 'm';
+        for(int i = 0; i < 3; i++) {
+            for (int j = 0; j < 3; j++) {
+                System.out.print(grid[i][j] + " ");
+            }
+            System.out.println("");
+        }
+        Assignment3 a = new Assignment3(grid, dictionary);
+        Set<String> s = a.manyDFSs();
+                System.out.println(s.size());
+        Iterator<String> it = s.iterator();
+        StringBuilder bigString = new StringBuilder("");
+        while (it.hasNext()) {            
+            String currentString = it.next();
+            System.out.println(currentString);
+            bigString.append(currentString);
+            bigString.append(" ");
+        }
+        if(bigString.length()>= 1) {
+            bigString.deleteCharAt(bigString.length()-1);
+        }
+        assertEquals(bigString.toString(),"");
+        
+        
+    }
+     @Test
+    public void EmptyDictionaryAndGridTest(){
+        //we have nothing also in the grid and in the dictionary
+        Assignment3 b = new Assignment3();
+        Assignment3.MyDictionary dictionary =  b.new MyDictionary();
+        char[][] grid = new char[3][3];
+        
+        Assignment3 a = new Assignment3(grid, dictionary);
+        Set<String> s = a.manyDFSs();
+                System.out.println(s.size());
+        Iterator<String> it = s.iterator();
+        StringBuilder bigString = new StringBuilder("");
+        while (it.hasNext()) {            
+            String currentString = it.next();
+            System.out.println(currentString);
+            bigString.append(currentString);
+            bigString.append(" ");
+        }
+        if(bigString.length()>= 1) {
+            bigString.deleteCharAt(bigString.length()-1);
+        }
+        assertEquals(bigString.toString(),"");
+        
+        
+    }
+     @Test
+    public void ComplexTest(){
+       // TODO code application logic here
```

Comment:
What is this TODO? Were you planning to do something here?

Reply:
i think it doesn't mean anything. it was written when i started and i forgot to delete it.